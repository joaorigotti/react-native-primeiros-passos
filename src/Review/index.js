import React from 'react';
import {
  View,
  Text,
} from 'react-native';
import Title from '../Title';
import stylesheet from './styles';

const styles = stylesheet();
const Review = props => {
  return (
    <View>
      <View style={styles.container}>
        <Title size={16} bold>{props.name}</Title>
        <Text style={styles.review}>{props.text}</Text>
      </View>
    </View>
  );
}

export default Review