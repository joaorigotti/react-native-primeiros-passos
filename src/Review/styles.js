import { StyleSheet } from 'react-native';
import * as cs from '../CommonStyles';

const padTop = cs.BASE_PADDING * 4;
const padBottom = cs.BASE_PADDING + 5;

export default () => StyleSheet.create({
  container: {
    width: cs.SCREEN_WIDTH,
    paddingLeft: padTop,
    paddingRight: padTop,
    paddingTop: padBottom,
    paddingBottom: padBottom,
  },

  review: {
    fontFamily: cs.MAIN_FONT,
    fontSize: 14,
    color: cs.COLOR.gray30,
    marginTop: cs.BASE_MARGING - 5,
  }
});