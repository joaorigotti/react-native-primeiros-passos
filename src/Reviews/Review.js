import React from 'react';
import {
  StyleSheet,
  View,
  Text,
} from 'react-native';
import { getWidth } from '../Utils';
import Title from '../Title';

const Review = props => {
  return (
    <View>
      <View style={styles.container}>
        <Title size={16} bold>{props.name}</Title>
        <Text style={styles.review}>{props.text}</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: getWidth(),
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 10,
    paddingBottom: 10,
  },

  review: {
    fontFamily: 'RobotoSlab-Regular',
    fontSize: 14,
    color: '#4d4d4d',
    marginTop: 5,
  }
});

export default Review