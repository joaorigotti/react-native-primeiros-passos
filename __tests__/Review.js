import React from 'react';
import { shallow } from 'enzyme';
import Review from '../src/Review';

describe('<Review />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Review />
    );
    
    expect(wrapper).toMatchSnapshot();
  });
});