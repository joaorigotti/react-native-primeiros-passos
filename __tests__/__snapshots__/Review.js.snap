// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "paddingBottom": 10,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 10,
        "width": 750,
      }
    }
  >
    <Photo
      bold={true}
      size={16}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#4d4d4d",
          "fontFamily": "RobotoSlab-Regular",
          "fontSize": 14,
          "marginTop": 5,
        }
      }
    />
  </View>
</View>
`;
